File: 100-same-tree/same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        # If both trees are None, they are the same
        if not p and not q:
            return True
        # If one of the trees is None, they are not the same
        if not p or not q:
            return False
        # If the values of the current nodes do not match, they are not the same
        if p.val != q.val:
            return False
        # Check recursively if the left and right subtrees are the same
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
            
        
        

File: 100-same-tree/same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function (p, q) {

   
    // Base case: both nodes are null
    if (!p && !q) return true;

    // If one is null and the other isn't, not the same
    if (!p || !q) return false;

    // If values differ, not the same
    if (p.val !== q.val) return false;

    // Recursively check left and right subtrees
    return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);

};